src/algebra/monoid_algebra.lean

```diff
@@ -15,6 +16,12 @@ a convolution product. To mathematicians this structure is known as the "monoid
 i.e. the finite formal linear combinations over a given semiring of elements of the monoid.
 The "group ring" ℤ[G] or the "group algebra" k[G] are typical uses.
 
+In fact the construction of the "monoid algebra" makes sense when `G` is not even a monoid, but
+merely a magma, i.e., when `G` carries a multiplication which is not required to satisfy any
+conditions at all. In this case the construction yields a not-necessarily-unital,
+not-necessarily-associative algebra but it is still adjoint to the forgetful functor from such
+algebras to magmas, and we prove this as `monoid_algebra.lift_magma`.
```

Comment:
```suggestion
not-necessarily-associative algebra (one characterized by `[non_unital_non_assoc_semiring A]`
`[smul_comm_class R A A]` `[is_scalar_tower R A A]` instead of `[semiring A]` `[algebra R A]`)
but it is still adjoint to the forgetful functor from such algebras to magmas, and we prove this as
`monoid_algebra.lift_magma` (and the corresponding version for additive magmas
`add_monoid_algebra.lift_magma`).
```

Reply:
I'm not sure if this is helpful. I worry about drowning the reader in detail and after all, if they really need this, it's available in the type of `lift_magma`.

Reply:
And likewise, with the `add_monoid_algebra` point. It's certainly subjective and I'll happily accept the suggestion if you feel strongly, but I feel it's better to explain each point on its own and allow the consumer to assemble them to the particular situation they need.

Reply:
Fair enough